.github/workflows/github-actions-test-boost-version.yml

```diff
@@ -0,0 +1,96 @@
+name: Test Boost Version
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      boostVersion:
+        description: 'Boost Version'
+        required: true
+        default: '1.81.0' 
+        type: string
+      boostChecksum:
+        description: 'Boost Checksum'
+        required: true
+        default: '4bf02e84afb56dfdccd1e6aec9911f4b' 
+        type: string
+  push:
+    paths:
+      - 'etc/DependencyInstaller.sh'
+      - '.github-actions-test-boost-version.yml'
+  pull_request:
+    paths:
+      - 'etc/DependencyInstaller.sh'
+      - '.github-actions-test-boost-version.yml'
+  schedule:
+  - cron: "0 8 * * *"
+
+jobs:  
+  testBoostVersionInput:
+    if: github.event_name == 'workflow_dispatch'
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-20.04", "ubuntu-22.04"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          submodules: recursive
+      - name: Run installer
+        shell: bash
+        run: |
+          sudo ./etc/DependencyInstaller.sh -boostVersion=${{ inputs.boostVersion }} -boostChecksum=${{ inputs.boostChecksum }}
+      - name: Build project
+        shell: bash
+        run: |
+          sudo ./etc/Build.sh
+      - name: Test build
+        shell: bash
+        run: |
+          build/src/openroad -help
+          sudo ./test/regression
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-results
+          path: |
+            src/**/test/results/diffs
+            src/**/test/results/*.log
+  testBoostVersionMatrix:
+    if: github.event_name != 'workflow_dispatch'
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-20.04", "ubuntu-22.04"]
+        boost: [{version: "1.78.0", checksum: "c2f6428ac52b0e5a3c9b2e1d8cc832b5"}, {version: "1.79.0", checksum: "e9213a48f0b19b7e01597c79f9128dd3"}, {version: "1.80.0", checksum: "077f074743ea7b0cb49c6ed43953ae95"}, {version: "1.81.0", checksum: "4bf02e84afb56dfdccd1e6aec9911f4b"}]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          submodules: recursive
+      - name: Run installer
+        shell: bash
+        run: |
+          sudo ./etc/DependencyInstaller.sh -boostVersion=${{ matrix.boost.version }} -boostChecksum=${{ matrix.boost.checksum }}
+      - name: Build project
+        shell: bash
+        run: |
+          sudo ./etc/Build.sh -threads=3
+      - name: Test build
+        shell: bash
+        run: |
+          build/src/openroad -help
+          export OMP_NUM_THREADS=3
+          sudo ./test/regression
```

Comment:
did this had any effect in the end or can we remove? also could you give it a try again without sudo?